### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by diegoarvz4**
## Milestone 2: Basic Redux Setup and Implementation.
For this Milestone we implemented the followint:

- We added the BookList and BookForm Components.
- We worked on our reducers using TDD.
- We connected the necesary state to the components that will be making use of it using the react-redux connect function.
- We made sure to pass the store with the provider, so that we could have access throughout the app.
-

### CODE REVIEW

File: src/components/BooksForm/BooksForm.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import uuid from 'uuid';
+
+export default () => {
```
Comment (SunnyAsar) [inline src/components/BooksForm/BooksForm.jsx]: moving forward, this should be a class components so we can hold state of the form

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default {
+  addBook: (state, action) => (
+    {
+      ...state,
+      books: [...state.books, action.payload],
+    }
```
Comment (SunnyAsar) [inline src/actions/index.js]: This kind of work belongs to the reducer an action should look like:
```
const addBook = book => ({
  type: "CREATE_BOOK",
  book,
});

```

The reducer should be what has access to the store to add or remove data based on the info provided by action.

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
# Hello @diegoarvz4  and @jofer86  👋 
 well done on the milestone so far 👏 

**For the finish**
- Let us update our `readme` to be professional 
- please find below in the comments the updates that are needed for the `actions/index` and its relationship with the reducer.

**Please request another review when the updates are in place**
Happy Coding

File: src/components/BooksForm/BooksForm.jsx (hunk 1 above)
Comment (jofer86) [reply src/components/BooksForm/BooksForm.jsx]: Resolved the above mentioned issue and created an Stateful component, which for the time being it holds the categories array.

File: src/actions/index.js (hunk 2 above)
Comment (jofer86) [reply src/actions/index.js]: Made the necessary adjustments to make the actions look like what it was told to us.

**Comment by jofer86** [comment]
Requesting a follow up of the previous code review. Changes were implemented and tests were changed accordingly.

File: src/actions/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default {
+  addBook: 'ADD_BOOK',
+  removeBook: 'REMOVE_BOOK',
+};
```
Comment (shloch) [inline src/actions/actionTypes.js:4]: nice strategy, i almost missed this 😃😃

**Review by shloch — APPROVED** [review]
## Nice job @diegoarvz4 🎅🎅🎅

### You can merge this PR 

Happy coding....see you on next mile 😃🎅🎅🎅

**Pull request merged**